pkgs/development/python-modules/spacy/default.nix

```
@@ -54,9 +54,15 @@ buildPythonPackage rec {
   # '';
 
   postPatch = ''
-    substituteInPlace setup.cfg --replace "thinc==7.4.1" "thinc>=7.4.1,<8"
+    substituteInPlace setup.cfg \
+      --replace "catalogue>=0.0.7,<1.1.0" "catalogue>=0.0.7,<2.1.0" \
+      --replace "plac>=0.9.6,<1.2.0" "plac>=0.9.6,<1.3.0" \
+      --replace "srsly>=1.0.2,<1.1.0" "srsly>=1.0.2,<2.1.0" \
+      --replace "thinc==7.4.1" "thinc>=7.4.1,<8"
```

so we don't have to revisit this as often
```suggestion
    substituteInPlace setup.cfg \
      --replace "catalogue>=0.0.7,<1.1.0" "catalogue>=0.0.7,<3.0" \
      --replace "plac>=0.9.6,<1.2.0" "plac>=0.9.6,<2.0" \
      --replace "srsly>=1.0.2,<1.1.0" "srsly>=1.0.2,<3.0" \
      --replace "thinc==7.4.1" "thinc>=7.4.1,<8"
```

---

I considered this, but I am worried that this will at some point introduce incompatible versions. For most of the spaCy ecosystem packages, we have disabled tests because they don't work with sandboxing, so builds may not show incompatibilities. Also, the spaCy people always pick fairly specific versions, even if they have newer versions of packages out, which seems to suggest that they anticipate incompatibilities (and do not follow semver).

At some point I had a NixOS test for spaCy that loaded the small English model and verified the output of some sentences to known-good output. I guess adding such a test could be a good way to have looser bounds while still having some guarantees that it actually works.

---

not sure, but a minimal use case should be selected.

for importable python libraries, nixpkgs decides the version for you. This is currently what spacy is, so if spacy is to remain in nixpkgs, this is the reality.

For python applications, you can pin versions to your heart's content.

My suggestions are in line with semver. Hopefully the upstream packages respect it, otherwise we need tests to verify the package is still coherent.